javascript/modules/client/test/integration/client-addressList.test.ts

```diff
@@ -132,8 +130,9 @@ describe("Client", () => {
 
   describe("Client instances", () => {
     it("Should create a working client", async () => {
-      const ctx = new ContextPlugin(contextParams);
-      const client = new ClientAddressList(ctx);
+      const ctx = new Context(contextParams);
+      const ctxPlugin = ContextPlugin.fromContext(ctx)
+      const client = new ClientAddressList(ctxPlugin);
```

Comment:
Maybe this is more future proof, just a random question: if the current plugin context has no extra state now... does it make sense to keep a separate class? Probably it does, so that when future settings appear we don't need to resplit again

Reply:
Yes, this was the exact thought process behind it, maybe it does not make sense atm, but if we want to add properties that are unique to the plugins in the future we can do it without having to split them up again. If in the end we don't add any property, we can get rid of it and use the normal `Context`